src/p2p_networking.rs

```diff
@@ -0,0 +1,52 @@
+use crate::errors::RuntimeError;
+use crate::{AsyncRuntime, PeerManager};
+use bitcoin::secp256k1::PublicKey;
+use log::debug;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use std::task::Poll;
+use std::time::Duration;
+use tokio::time::sleep;
+
+pub fn connect_to_peer(
+    rt: &AsyncRuntime,
```

Comment:
```suggestion
    handle: &Handle,
```
We should have only one instance of the async runtime and pass handles to downstream functions.

Reply:
Moreover let's make this function `async`, let the caller to block on it if needed.

Reply:
Fixed. It is now being `blocked_on` in the lib.rs initialization flow.